unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -6936,129 +6936,149 @@ public bool objectIsCurrentlyVisible(SimObjPhysics sop, float maxDistance)
             Bounds objectBounds = new Bounds(
-                new Vector3(float.PositiveInfinity, float.PositiveInfinity, float.PositiveInfinity),
-                new Vector3(-float.PositiveInfinity, -float.PositiveInfinity, -float.PositiveInfinity)
+                center: new Vector3(float.PositiveInfinity, float.PositiveInfinity, float.PositiveInfinity),
+                size: new Vector3(-float.PositiveInfinity, -float.PositiveInfinity, -float.PositiveInfinity)
             );
             objectBounds.Encapsulate(theObject.transform.position);
             foreach (Transform vp in theObject.VisibilityPoints) {
                 objectBounds.Encapsulate(vp.position);
             }
             float fudgeFactor = objectBounds.extents.magnitude;
-
             List<Vector3> filteredPositions = positions.Where(
-                p => (Vector3.Distance(p, theObject.transform.position) <= maxVisibleDistance + fudgeFactor + gridSize)
+                p => (Vector3.Distance(a: p, b: theObject.transform.position) <= maxVisibleDistance + fudgeFactor + gridSize)
             ).ToList();
 
-            Dictionary<string, List<float>> goodLocationsDict = new Dictionary<string, List<float>>();
+            // set each key to store a list
+            Dictionary<string, List<object>> validAgentPoses = new Dictionary<string, List<object>>();
             string[] keys = {"x", "y", "z", "rotation", "standing", "horizon"};
             foreach (string key in keys) {
-                goodLocationsDict[key] = new List<float>();
+                validAgentPoses[key] = new List<object>();
             }
 
-            for (int k = (int)-30/action.horizon; k <= (int)60/action.horizon; k++) {
-                m_Camera.transform.localEulerAngles = new Vector3(action.horizon * k, 0f, 0f);
-                for (int j = 0; j < 2; j++) { // Standing / Crouching
-                    if (j == 0) {
-                        stand();
-                    } else {
-                        crouch();
-                    }
-                    for (int i = 0; i < 4; i++) { // 4 rotations
-                        transform.rotation = Quaternion.Euler(new Vector3(0.0f, 90.0f * i, 0.0f));
-                        foreach (Vector3 p in filteredPositions) {
-                            transform.position = p;
+            // iterate over each reasonable agent pose
+            foreach (float horizon in horizons) {
+                // recall that horizon=60 is look down 60 degrees and horizon=-30 is look up 30 degrees
+                if (horizon > maxDownwardLookAngle || horizon < -maxUpwardLookAngle) {
+                    errorMessage = $"Each horizon must be in [{-maxUpwardLookAngle}:{maxDownwardLookAngle}]";
+                    actionFinished(success: false);
+                }
+                m_Camera.transform.localEulerAngles = new Vector3(horizon, 0f, 0f);
+
+                foreach (bool standing in standings) {
```

Comment:
Note that we always treat `standing` as a boolean, instead of a float (0 or 1) because that is what's passed into TeleportFull and is most consistent with the rest of the physics controller.